### PR TITLE
Statistics estimation for values (v2)

### PR DESCRIPTION
**Pull request by losipiuk**
Supersedes https://github.com/Teradata/presto/pull/572

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java [hunk 1]
```diff
@@ -45,7 +47,9 @@ private StatsCalculatorTester(LocalQueryRunner queryRunner)
 
     private static LocalQueryRunner createQueryRunner()
     {
-        Session session = testSessionBuilder().build();
+        Session session = testSessionBuilder()
+                .setSystemProperty(USE_NEW_STATS_CALCULATOR, "true")
```
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java:50]: So we already have an old one? :)

File: presto-main/src/test/java/com/facebook/presto/cost/TestValuesNodeStats.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            throws Exception
+    {
+        tester = new StatsCalculatorTester();
+    }
+
+    @Test
+    public void testStatsForValuesNode()
+            throws Exception
+    {
+        tester.assertStatsFor(pb -> pb
```
Comment (sopel39) [inline presto-main/src/test/java/com/facebook/presto/cost/TestValuesNodeStats.java:41]: add test for strings and empty values node

File: presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java (hunk 1 above)
Comment (losipiuk) [reply presto-main/src/test/java/com/facebook/presto/cost/StatsCalculatorTester.java:50]: Yeah

**Comment by losipiuk** [comment]
merged
